packages/connect/src/relay/state.ts

```diff
@@ -90,6 +94,16 @@ class RelayState {
     context[source.toB58String()].update(toSource)
   }
 
+  /**
+   * Performs an operation for each relay context in the current set.
+   * @param action
+   */
+  async forEach(action: (dst: string, ctx: RelayContext) => void) {
+    await Promise.all(
+      Array.from(this.relayedConnections.values()).map((s) => Object.entries(s).map((e) => action(e[0], e[1])))
```

Comment:
Probably safer to use `nAtATime` from `@hoprnet/hopr-utils` to prevent from potential process locking